great_expectations/execution_engine/pandas_execution_engine.py

```diff
@@ -201,6 +237,26 @@ def get_batch_data_and_markers(
             buf.seek(0)
             df = reader_fn(buf, **reader_options)
 
+        elif isinstance(batch_spec, GCSBatchSpec):
+            if self._gcs is None:
+                raise ge_exceptions.ExecutionEngineError(
+                    f"""PandasExecutionEngine has been passed a GCSBatchSpec,
+                        but the ExecutionEngine does not have an GCS client configured. Please check your config."""
+                )
+            gcs_engine = self._gcs
+            gcs_url = GCSUrl(batch_spec.path)
+            reader_method: str = batch_spec.reader_method
+            reader_options: dict = batch_spec.reader_options or {}
+            gcs_bucket = gcs_engine.get_bucket(gcs_url.bucket)
+            gcs_blob = gcs_bucket.blob(gcs_url.blob)
+            logger.debug(
+                f"Fetching GCS blob. Bucket: {gcs_url.bucket} Blob: {gcs_url.blob}"
+            )
+            reader_fn = self._get_reader_fn(reader_method, gcs_url.blob)
+            buf = BytesIO(gcs_blob.download_as_bytes())
+            buf.seek(0)
+            df = reader_fn(buf, **reader_options)
+
```

Comment:
Identical to the processing done for `Azure` and `S3` with slight deviations in the specific SDK methods use to get the appropriate bucket/blob